core/tee/tee_svc_storage.c

```diff
@@ -1069,8 +1097,12 @@ TEE_Result syscall_storage_obj_write(unsigned long obj, void *data, size_t len)
 		goto exit;
 
 	o->info.dataPosition += len;
```

Comment:
Hmm, this is actually a problem with the way we're layering this. The update in the lower layer is already committed and what we would need now is unrolling it. This problem is out of scope of this P-R so please ignore my previous comment for now.